src/lint.cpp

```diff
@@ -327,6 +333,12 @@ void linter::declare_variable(scope &scope, identifier name, variable_kind kind,
                    declared, used_var.name,
                    /*is_assigned_before_declaration=*/false);
                break;
+             case used_variable_kind::use_and_assignment:
```

Comment:
I think this deserves a test. I think the following examples will execute this code path:
```js
function f() {
  x += 3;  // (no error)
}
let x = 1;
```
```js
function f() {
  x += 3;  // ERROR: assignment to const variable 'x'
}
const x = 1;
```

Reply:
Works fine